Common/Data/Consolidators/WickedRenkoConsolidator.cs

```diff
@@ -0,0 +1,277 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Data.Consolidators
+{
+    /// <summary>
+    /// This consolidator can transform a stream of <see cref="BaseData"/> instances into a stream of <see cref="RenkoBar"/>
+    /// with Renko type <see cref="RenkoType.Wicked"/>.
+    /// </summary>
+    public class WickedRenkoConsolidator : IDataConsolidator
+    {
+        private DateTime _closeOn;
+        private decimal _closeRate;
+        private bool _firstTick = true;
+        private decimal _highRate;
+        private RenkoBar _lastWicko;
+        private decimal _lowRate;
+        private DateTime _openOn;
+        private decimal _openRate;
+        private decimal _barSize;
+        private DataConsolidatedHandler _dataConsolidatedHandler;
+        private RenkoBar _currentBar;
+        private IBaseData _consolidated;
+
+        // Used for unit tests
+        internal RenkoBar OpenRenkoBar =>
+            new RenkoBar(null, _openOn, _closeOn, _barSize, _openRate, _highRate, _lowRate, _closeRate);
```

Comment:
Could we directly remove `OpenRenkoBar` and use `WorkingData` instead?

Reply:
This is interesting. From what I see, this is a means of bypassing the fact that for the end user, assignment of `_currentBar` occurs upon `OnDataConsolidated`. For a property like `WorkingData`, I reckon its best to change each of `_openOn, _closeOn, _barSize, _openRate, _highRate, _lowRate, _closeRate` to `internal` than to determine what is returned based on whether or not the caller is a test. Doing this and then adding a method in WickedConsolidatorTests which returns the same value as `OpenRenkoBar` can allow removal of `OpenRenkoBar`.